pkg/resourceprovider/provider_test.go

```diff
@@ -255,6 +256,29 @@ var _ = Describe("Resource Metrics Provider", func() {
 		))
 	})
 
+	It("should batch pods for queries by sets of size 500 each", func() {
```

Comment:
It would be good to add more tests for batching logic e.g.
- one pod per ns,
- only one pod,
- number of pod equal to exactly the batch size

We could add more tests in the follow-up PRs.